design/one-pager-stack-relationship-labels.md

```diff
@@ -0,0 +1,165 @@
+# Stack Parent/Child Relationship Label Spec
+
+- Owner: Steven Rathbauer (@rathpc)
+- Reviewers: Crossplane Maintainers
+- Status: Draft
+
+## Proposal
+
+[crossplaneio/crossplane#934](https://github.com/crossplaneio/crossplane/issues/934)
+
+## Problem
+
+There is currently no easy way to get a list of all resources that are owned by a parent resource.
+We can leverage CRD labels to make this easier. As long as we have a standard format for doing so.
+
+Per the proposal, it seems that the closest example of what I am referring to is shown on this page
+https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/#labels within the
+`app.kubernetes.io/part-of` label. However since we are not exactly using these labels for the
+intentions defined within that page, we should use something following the patterns instead.
+
+As long as we define this label on all children CRD's, we should then be able to query all children
+of a given parent name using those labels.
+
+## Design
+
+Wordpress example file tree structure of the example files below:
+
+```text
+stack.yaml
+|
+`-- crd.yaml
+    |
+    `-- resource.yaml
+        |-- kubernetesapplication.yaml
+        |-- kubernetescluster.yaml
+        |-- mysqlinstance.yaml
+        |-- ...
+```
+
+### Example wordpress `stack.yaml`
+
+```yaml
+apiVersion: stacks.crossplane.io/v1alpha1
+kind: Stack
+metadata:
+  creationTimestamp: "2019-10-23T23:47:38Z"
+  generation: 1
+  name: sample-stack-wordpress
+  namespace: app-project1-dev
+  ownerReferences:
+  - apiVersion: stacks.crossplane.io/v1alpha1
+    kind: StackInstall
+    name: sample-stack-wordpress
+    uid: 86c89c94-cfc0-474e-b3b3-beffc09e7793
+  resourceVersion: "6431"
+  selfLink: /apis/stacks.crossplane.io/v1alpha1/namespaces/app-project1-dev/stacks/sample-stack-wordpress
+  uid: ec52c8c2-379f-45ec-9458-e40f070f8d2e
+  labels:
+    app.kubernetes.io/managed-by: stack-manager
```

Comment:
Nitpick: theoretically, the `Stack` **could** be considered a child of the `StackInstall` or `ClusterStackInstall` object, so we may include that in the labels in the future.

Reply:
I agree. Although, for the purposes of showing an example I think this is sufficient so I will leave that out for now.